src/Illuminate/Database/Query/Builder.php

```diff
@@ -2264,6 +2264,23 @@ public function toSql()
         return $this->grammar->compileSelect($this);
     }
 
+    /**
+     * Get the SQL representation of the query replacing all bindings.
+     *
+     * @return string
+     */
+    public function toSqlWithBindings()
+    {
+        return Str::replaceArray(
+            '?',
+            collect($this->connection->prepareBindings($this->getBindings()))
+                ->map(function ($i) {
+                    return (is_string($i)) ? "'{$i}'" : $i;
```

Comment:
I guess it would be safer to call some existing functionality to quote query parameters, otherwise this would not work when one of the bound strings contains a single quote.

Reply:
@thiagorb I think PDO `quote` method solve this problem. Can you check it again?